models/validateEntity.go

```diff
@@ -125,22 +137,31 @@ func validateParent(ent string, entNum int, t map[string]interface{}) (map[strin
 
 	default:
 		parentInt := u.GetParentOfEntityByInt(entNum)
-		parent := u.EntityToString(parentInt)
+		parentStr := u.EntityToString(parentInt)
 
-		ctx, cancel := u.Connect()
-		if GetDB().Collection(parent).
-			FindOne(ctx, bson.M{"_id": objID}).Err() != nil {
+		p, err := GetEntity(req, parentStr)
+		if len(p) > 0 {
+			parent["parent"] = parentStr
+			parent["hierarchyName"] = getHierarchyName(p)
+			return parent, true
+		} else if err != "" {
 			println("ENTITY VALUE: ", ent)
 			println("We got Parent: ", parent, " with ID:", t["parentId"].(string))
 			return u.Message(false,
 				"ParentID should correspond to Existing ID"), false
-
 		}
-		defer cancel()
 	}
 	return nil, true
 }
 
+func getHierarchyName(parent map[string]interface{}) string {
+	if parent["hierarchyName"] != nil {
+		return parent["hierarchyName"].(string)
+	} else {
+		return parent["name"].(string)
+	}
+}
+
 func validateJsonSchema(entity int, t map[string]interface{}) (map[string]interface{}, bool) {
```

Comment:
As a potential feauture request, I suggest to store compiled schema JSONs at init in a map variable so that we don't have to open a file and compile everytime

Reply:
The file is opened and compiled just once at init and added to the schema compiler at loadJsonSchemas